tests/integration/targets/ec2_vpc_dhcp_option/tasks/main.yml

```diff
@@ -0,0 +1,897 @@
+  - name: test a DHCP option exists (check mode)
+    ec2_vpc_dhcp_option:
+      state: present
+      vpc_id: "{{ vpc_id }}"
+    register: found_dhcp_options
+    check_mode: true
+
+  - assert:
+      that:
+        # FIXME: options have to be provided to match the option associated with the VPC
+        - not found_dhcp_options.changed
+        - not found_dhcp_options.new_options
+
+  # FIXME: always reassociated when vpc_id is provided without options, so here we provide the default options
+  - name: test a DHCP option exists
+    ec2_vpc_dhcp_option:
+      state: present
+      vpc_id: "{{ vpc_id }}"
+      domain_name: "{{ aws_domain_name }}"
+      dns_servers:
+        - AmazonProvidedDNS
+      tags:
+        Name: "{{ resource_prefix }}"
+    register: found_dhcp_options
+
+  - assert:
+      that:
+        - found_dhcp_options is not changed
+        - found_dhcp_options.dhcp_options_id is defined
+        - not found_dhcp_options.changed or dhcp_options is defined and dhcp_options.dhcp_options_id == found_dhcp_options.dhcp_options_id
+
+  # Create a DHCP option set that inherits from the default set and does not delete the old set
+
+  # FIXME: check mode causes a traceback
+  #- name: create a DHCP option set that inherits from the default set (check mode)
+  #  ec2_vpc_dhcp_option:
+  #    state: present
+  #    vpc_id: "{{ vpc_id }}"
+  #    inherit_existing: True
+  #    ntp_servers:
+  #        - 10.0.0.2
+  #        - 10.0.1.2
+  #    netbios_name_servers:
+  #        - 10.0.0.1
+  #        - 10.0.1.1
+  #    netbios_node_type: 2
+  #    delete_old: False
+  #  register: dhcp_options
+  #  check_mode: true
+
+  #- assert:
+  #    that:
+  #      - dhcp_options.changed
+
+  - name: create a DHCP option set that inherits from the default set
+    ec2_vpc_dhcp_option:
+      state: present
+      vpc_id: "{{ vpc_id }}"
+      inherit_existing: True
+      ntp_servers:
+          - 10.0.0.2
+          - 10.0.1.2
+      netbios_name_servers:
+          - 10.0.0.1
+          - 10.0.1.1
+      netbios_node_type: 2
+      delete_old: False
+    register: dhcp_options
+
+  - debug: msg="{{ dhcp_options.new_options['netbios-node-type'] }}"
+
+  - assert:
+      that:
+        - dhcp_options.changed
+        - dhcp_options.new_options
+        - dhcp_options.new_options.keys() | list | sort == ['domain-name', 'domain-name-servers', 'netbios-name-servers', 'netbios-node-type', 'ntp-servers']
+        - dhcp_options.new_options['ntp-servers'] | sort == ['10.0.0.2', '10.0.1.2']
+        - dhcp_options.new_options['netbios-name-servers'] | sort == ['10.0.0.1', '10.0.1.1']
+        # FIXME: module/aws randomly returns as a string or list
+        - dhcp_options.new_options['netbios-node-type'] in ['2', ['2']]
+        # found I think is false, because the options are different and no id is provided?
+        - dhcp_options.new_options['domain-name'] == "{{ aws_domain_name }}"
```

Comment:
`dhcp_options.new_options['domain-name']` is a list. This is probably the reason why the test fail here. Is this expected?

Reply:
cf. https://github.com/ansible-collections/amazon.aws/blob/main/plugins/modules/ec2_vpc_dhcp_option.py#L325

Reply:
It depends. :)  Thanks, I thought I caught all of these.